pageserver/src/tenant/layer_cache.rs

```diff
@@ -0,0 +1,143 @@
+use super::storage_layer::{PersistentLayer, PersistentLayerDesc, PersistentLayerKey, RemoteLayer};
+use super::Timeline;
+use crate::tenant::layer_map::{self, LayerMap};
+use anyhow::Result;
+use std::sync::{Mutex, Weak};
+use std::{collections::HashMap, sync::Arc};
+
+pub struct LayerCache {
```

Comment:
Could add some type docs, like:

- it's meant to facilitate mapping to/from whatever PersistentLayerDesc::key type to an actual in-memory layer

Is it meant that eventually we will have compaction and gc in here as well?

Reply:
compaction and GC will not be there because they need to modify the layer map (remove / add some layers). But download and eviction will be there. Added in be4999713a851e2f005cb65c5bd755f5be9eafcf